alf/environments/wrappers.py

```diff
@@ -102,3 +104,200 @@ def step(self, action):
 
     def reset(self, **kwargs):
         return self.env.reset(**kwargs)
+
+    def __getattr__(self, name):
+        """Forward all other calls to the base environment."""
+        return getattr(self.env, name)
+
+
+@gin.configurable
+class DMAtariPreprocessing(gym.Wrapper):
+    """
+    Derived from tf_agents AtariPreprocessing. Three differences:
+    1. If terminating on losing a life, stepping a NO-OP action instead of resetting
+    2. Random number of NOOPs after reset
+    3. FIRE after step 2
+
+    To see a complete list of atari wrappers used by DeepMind, see
+    https://github.com/ray-project/ray/blob/master/python/ray/rllib/env/atari_wrappers.py
+    Also see OpenAI Gym's implementation (not completely the same):
+    https://github.com/openai/gym/blob/master/gym/wrappers/atari_preprocessing.py
+
+    Note: This wrapper does not handle framestacking. It can be paired with FrameStack.
+    See ac_breakout.gin for an example.
+    """
+    def __init__(self,
+                 env,
+                 frame_skip=4,
+                 noop_max=30,
+                 terminal_on_life_loss=False,
+                 screen_size=84):
+        """Constructor for an Atari 2600 preprocessor.
+
+        Args:
```

Comment:
comment should follow the format of other comment in alf.

Reply:
+1

Reply:
updated